components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/store/InMemoryBasedTemplateManager.java

```diff
@@ -0,0 +1,263 @@
+ */
+public class InMemoryBasedTemplateManager implements TemplatePersistenceManager {
+
+    private final Map<String, Map<String, NotificationTemplate>> defaultEmailTemplates;
+    private final Map<String, Map<String, NotificationTemplate>> defaultSMSTemplates;
+
+    /**
+     * Initializes the in-memory template manager by populating default email and SMS templates.
+     */
+    public InMemoryBasedTemplateManager() {
+
+        defaultEmailTemplates = populateTemplates(I18nMgtDataHolder.getInstance().getDefaultEmailTemplates());
+        defaultSMSTemplates = populateTemplates(I18nMgtDataHolder.getInstance().getDefaultSMSTemplates());
+    }
+
+    @Override
+    public void addNotificationTemplateType(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        // This method does nothing cause addition is not allowed for in-memory template manager.
+    }
+
+    /**
+     * Checks if a notification template type exists in the in-memory store.
+     *
+     * @param displayName         The display name of the template type.
+     * @param notificationChannel The channel of the notification (e.g., Email, SMS).
+     * @param tenantDomain        The tenant domain.
+     * @return true if the template type exists, false otherwise.
+     * @throws NotificationTemplateManagerServerException if an error occurs during the operation.
+     */
+    @Override
+    public boolean isNotificationTemplateTypeExists(String displayName, String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        if (!StringUtils.isBlank(displayName)) {
+            if (NotificationChannels.SMS_CHANNEL.getChannelType().equals(notificationChannel)) {
+                return defaultSMSTemplates.containsKey(displayName.toLowerCase());
+            }
+            return defaultEmailTemplates.containsKey(displayName.toLowerCase());
+        }
+        return false;
+    }
+
+    /**
+     * Lists all notification template types for a given channel in the in-memory store.
+     *
+     * @param notificationChannel The channel of the notification (e.g., Email, SMS).
+     * @param tenantDomain        The tenant domain.
+     * @return A list of template type display names.
+     * @throws NotificationTemplateManagerServerException if an error occurs during the operation.
+     */
+
+    @Override
+    public List<String> listNotificationTemplateTypes(String notificationChannel, String tenantDomain)
+            throws NotificationTemplateManagerServerException {
+
+        List<String> displayNames = new ArrayList<>();
+        Map<String, Map<String, NotificationTemplate>> defaultTemplates =  getTemplateMap(notificationChannel);
+        defaultTemplates.forEach((displayName, innerMap) ->
+                innerMap.forEach((locale, template) ->
+                        displayNames.add(template.getDisplayName())
```

Comment:
If there are two templates for same type (like in two locale), aint this cased to return with a list that have  duplicate template types records?

Reply:
It does not make a difference, hence we return after removing the duplicates and merging [1].

[1] https://github.com/wso2-extensions/identity-event-handler-notification/pull/256/files#diff-55202b63368a9acaf49ac62f2c542327e0302c40fd60374d7378a6dcaaafd511R200-R210